src/OpenLoco/src/World/CompanyAi.cpp

```diff
@@ -812,12 +816,102 @@ namespace OpenLoco
     {
     }
 
+    // 0x00488563
+    static void removeCompanyTracksRoadsOnTile(CompanyId id, World::TilePos2 tilePos)
+    {
+        bool reTryTile = true;
+        while (reTryTile)
+        {
+            reTryTile = false;
+
+            auto tile = World::TileManager::get(tilePos);
+            for (auto& el : tile)
+            {
+                auto* elTrack = el.as<World::TrackElement>();
+                auto* elRoad = el.as<World::RoadElement>();
+                if (elTrack != nullptr)
+                {
+                    if (elTrack->owner() != id)
+                    {
+                        continue;
+                    }
+                    reTryTile = true;
+
+                    GameCommands::TrackRemovalArgs args{};
```

Comment:
Considering the two game commands are so alike, perhaps you extract getting track/road arguments into a templated function? That should reduce/remove the duplication in the two blocks here, and make the function more readable.

Reply:
On second thought, the field names are slightly different, making this more complicated than I originally thought. I don't want to move the goal posts too much, so let's just leave this be.